src/recipes.jl

```diff
@@ -0,0 +1,36 @@
+using RecipesBase
+
+#-----------------------------------------------------------------# partitionplot
+@userplot struct PartitionPlot
+    args
+end
+
+@recipe function f(tp::PartitionPlot; nparts = 50, stat = nothing, by = nothing)
+    t = tp.args[1]
+    selection = tp.args[2]
+    o = (stat == nothing) ? 
+            make_stat(fieldtype(eltype(t), IndexedTables.colindex(t, selection))) : 
+            stat 
+    if by == nothing 
+        @series begin 
+            title --> selection
+            reduce(Partition(o, nparts), t; select = selection)
+        end
+    else
+        out = collect(groupreduce(Partition(o, nparts), t, by; select = selection))
+        layout --> length(out)
+        for i in 1:length(out)
+            @series begin 
+                title --> string(selection) * " ($(out[i][1]))"
+                subplot --> i
```

Comment:
Do you need this? I thought the `series` would dispose on the correct `subplot` automatically. Also, it'd be nice to have it in such a way that if the user calls `partitionplot(t, :x, by = :y, layout = 1)`  the `series` would be all in the same subplot but in different colours. You should also be using `label` instead of `title` in that case. You should be able to use `if` conditions inside a `@series` macro, so:

```julia
if :layout in keys(plotattributes) && plotattributes[:layout] < length(out)
    label --> string(selection) * " ($(out[i][1]))"
else
    title --> string(selection) * " ($(out[i][1]))"
    subplot --> i
```

Reply:
Huh, you're right.  At some point I was doing something similar and the Value/Parts series alternating across subplots incorrectly.  I don't remember exactly what I was doing.  I may just remove the title and put them into the label.

Reply:
Now that I'm thinking about it, there's no guarantee that plotting them in a single plot makes sense since the observations may not line up.  `Partition` only tracks a Y variable.  It's on my to-do list to make a similar type that tracks a Y and an X variable.

Reply:
I see... So either the user should provided explicitly a partition binning or you should somehow force the algorithm to return a partition binnning that is equal across conditions?

Reply:
Partition sizes/not lining up isn't an issue.  I'm seeing two problems: 1) You want to compare groups using x-axis "% of nobs" rather than "nobs" unless nobs is equal across all groups, and 2) groups could be distributed differently over an index variable and that information gets lost.